### PR TITLE
[vcpkg] Update hint message

### LINKED ISSUE #4749: Message to re-bootstrap vcpkg does not take host platform into account
Even when on a Linux machine, it will display
>Warning: Different source is available for vcpkg (0.0.113 -> 2018.10.20). Use .\bootstrap-vcpkg.bat to update.

We should `#ifdef` on the host machine and display `.sh` for non-Windows

### PR DESCRIPTION
**Pull request by PhoebeHui**
Fixes https://github.com/microsoft/vcpkg/issues/4749 and https://github.com/microsoft/vcpkg/issues/10910

I didn't find the message for '.\vcpkg edit' which mentioned in issue #10910, it should be removed.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]
Thanks for your contribution! Will merge as soon as CI passes.

File: toolsrc/src/vcpkg/update.cpp [hunk 1]
```diff
@@ -74,13 +74,20 @@ namespace vcpkg::Update
...
             System::print2("\n"
                            "To update these packages and all dependencies, run\n"
-                           "    .\\vcpkg upgrade\n"
+                           "  %s\n"
                            "\n"
                            "To only remove outdated packages, run\n"
                            "    .\\vcpkg remove --outdated\n"
-                           "\n");
+                           "\n",
+                           vcpkg_upgrade_cmd);
```
Comment (ras0219) [inline toolsrc/src/vcpkg/update.cpp:90]: Missed `.\\vcpkg remove --outdated` -- I'd recommend putting just the `.\\vcpkg` part into a variable, then using `%s upgrade` and `%s remove --outdated`.

Comment (PhoebeHui) [reply toolsrc/src/vcpkg/update.cpp]: Thanks for your review!

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**